src/main/java/randoop/generation/constanttfidf/ConstantMiningStatistics.java

```diff
@@ -0,0 +1,272 @@
+    if (GenInputsAbstract.literals_level == ClassLiteralsMode.CLASS) {
+      throw new RuntimeException("Should not update numClassesWith in CLASS level");
+    }
+    scopeStatisticsMap
+        .computeIfAbsent(getScope(type), __ -> new ScopeStatistics())
+        .addClassesWith(seq, numClassesWithConstant);
+  }
+
+  /**
+   * Add and update the numClasses of the current scope.
+   *
+   * @param type the type of the class
+   * @param numClasses the total number of classes in the current scope
+   */
+  public void addToTotalClasses(Object type, int numClasses) {
+    if (GenInputsAbstract.literals_level == ClassLiteralsMode.CLASS) {
+      throw new RuntimeException("Should not update totalClasses in CLASS level");
+    }
+    scopeStatisticsMap
+        .computeIfAbsent(getScope(type), __ -> new ScopeStatistics())
+        .addToTotalClasses(numClasses);
+  }
+
+  /**
+   * Get all sequences that had been recorded under the specific scope, which are the constants
+   * extracted by constant mining.
+   *
+   * @param scope the specific package, class, or the "all" scope
+   * @return the set of sequences that have been recorded under the specific scope
+   */
+  public Set<Sequence> getSequencesForScope(Object scope) {
+    ScopeStatistics stats = scopeStatisticsMap.get(scope);
+    if (stats == null) {
+      Log.logPrintf("Scope %s is not a key in scopeStatisticsMap%n", scope);
+      return new HashSet<>();
+    }
+
+    return stats.getSequenceSet();
+  }
+
+  /**
+   * Get the frequency information for all scopes.
+   *
+   * @return the frequency information for all scopes
+   */
+  public Map<Object, Map<Sequence, Integer>> getNumUses() {
+    Map<Object, Map<Sequence, Integer>> res = new HashMap<>();
+    scopeStatisticsMap.forEach((key, value) -> res.put(key, value.getNumUses()));
+    return res;
+  }
+
+  /**
+   * Get the frequency information of the given scope.
+   *
+   * @param scope a type, a package, or the "all" scope
+   * @return the frequency information of the given scope
+   */
+  public Map<Sequence, Integer> getNumUses(Object scope) {
+    return scopeStatisticsMap.get(scope).getNumUses();
+  }
+
+  /**
+   * Get the numClassesWith information for all scopes.
+   *
+   * @return the numClassesWith information for all scopes
+   */
+  public Map<Object, Map<Sequence, Integer>> getNumClassesWith() {
+    Map<Object, Map<Sequence, Integer>> res = new HashMap<>();
+    scopeStatisticsMap.forEach((key, value) -> res.put(key, value.getNumClassesWith()));
+    return res;
+  }
+
+  /**
+   * Get the numClassesWith information of the given scope.
+   *
+   * @param scope a scope
+   * @return the numClassesWith information of the given scope
+   */
+  public Map<Sequence, Integer> getNumClassesWith(Object scope) {
+    if (GenInputsAbstract.literals_level == ClassLiteralsMode.CLASS) {
+      throw new RandoopBug("Should not get numClassesWith in CLASS level");
+    }
+    return getNumClassesWith().get(getScope(scope));
```

Comment:
It does not make sense to call `getScope` on a scope.  Is the `@param` documentation wrong, or is the call to `getScope()` wrong?  I made guesses elsewhere, but I might have guessed wrong.